sstables/storage.cc

```diff
@@ -533,6 +540,28 @@ future<> s3_storage::wipe(const sstable& sst, sync_dir) noexcept {
     co_await sys_ks.sstables_registry_delete_entry(_location, sst.generation());
 }
 
+future<> s3_storage::remove_by_registry_entry(utils::UUID uuid, entry_descriptor desc) {
+    auto prefix = format("/{}/{}", _bucket, uuid);
```

Comment:
to avoid adding another interface, which we'll never implement with fs mode,

what if we teach s3_storage::wipe() to load toc content if _recognized_components is empty(). and sstable_directory::system_keyspace_components_lister::garbage_collect() builds a sstable object with sstables::entry_descriptor (without actually loading it), which would be enough for wiping purposes. is it worth it?

and things are simplified, as we confine things like _sys_ks.sstables_registry_delete_entry(_location, gen); (which is used in wipe()) to s3_storage impl.